mc2mc/mc2mc.go

```diff
@@ -72,37 +79,85 @@ func mc2mc(envs []string) error {
 			query.WithPartitionValue(cfg.DevEnablePartitionValue == "true"),
 			query.WithColumnOrder(),
 		).Build()
+		if err != nil {
+			return errors.WithStack(err)
+		}
+		queriesToExecute = append(queriesToExecute, queryToExecute)
 	case "REPLACE":
-		queryToExecute, err = query.NewBuilder(
+		queryBuilder := query.NewBuilder(
 			l,
 			client.NewODPSClient(l, cfg.GenOdps()),
-			queryToExecute,
 			query.WithMethod(query.REPLACE),
 			query.WithDestination(cfg.DestinationTableID),
-			query.WithOverridedValue("_partitiontime", fmt.Sprintf("TIMESTAMP('%s')", dstart)),
-			query.WithOverridedValue("_partitiondate", fmt.Sprintf("DATE(TIMESTAMP('%s'))", dstart)),
 			query.WithAutoPartition(cfg.DevEnableAutoPartition == "true"),
 			query.WithPartitionValue(cfg.DevEnablePartitionValue == "true"),
 			query.WithColumnOrder(),
-		).Build()
+		)
+
+		// generate queries for each date
+		// if it contains break marker, it must uses window range greather than 1 day
+		// if table destination is partition table, then it will be replaced based on the partition date
+		// for non partition table, only last query will be applied
```

Comment:
for the non partitioned-table, are we planning to just use the last query or we will still execute everything an let the last one replace the previous ones?

Reply:
we still execute everything and let the last one replace the previous ones

Reply:
we can improve later if there's a performance degradation